.github/workflows/build-test-deploy.yml

```diff
@@ -0,0 +1,75 @@
+name: Build, test, and deploy
+
+on:
+  # Run this workflow every day at 3:00 UTC to update caches.
+  schedule: [cron: '0 3 * * *']
+  # Allow manual trigger.
+  workflow_dispatch:
+  pull_request:
+    branches: ["master"]
+  push:
+    branches: ["master"]
+    tags: ["v*"]
+
+# Cancel old PR builds when pushing new commits.
+concurrency:
+  group: build-test-deploy-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  #############################################################
+  # Preparation jobs
+  #############################################################
+
+  prepare:
+    name: Prepare
+    uses: iphydf/ci-tools/.github/workflows/deploy-prepare.yml@deploy-wf
```

Comment:
its merged now

Reply:
This now needs https://github.com/TokTok/ci-tools/pull/97

Reply:
Done.